deep_qa/data/instances/sequence_tagging/verb_semantics_instance.py

```diff
@@ -0,0 +1,173 @@
+from typing import Dict, List, Tuple
+
+import numpy
+from overrides import overrides
+
+from ..instance import TextInstance, IndexedInstance
+from ...data_indexer import DataIndexer
+
+
+class VerbSemanticsInstance(TextInstance):
+    """
+    A VerbSemanticsInstance is a :class:`TextInstance` that is a single sentence of text,
+    where that sentence has some associated categorical label.
+    """
+    def __init__(self, sentence: List[str], verb: (int, int), entity: (int, int), state_change: str, arg1: (int, int), arg2: (int, int), index: int=None):
+        """
+        sentence: a sentence from which we want to predict verb semantics.
+        A sentence can mention multiple verbs and entities, a data instance
+        is talking about a particular (verb, entity) pair in a given sentence
+        verb: span of the verb indicating the event being referred to in the query
+        entity: entity indicating 
+        """
+        """ TODO: initialize all members of the class"""
+        self.sentence = sentence
+        self.verb = verb
+        self.entity = entity
+        tags = self.__make_tag_sequence(arg1, arg2)
+        super(VerbSemanticsInstance, self).__init__((state_change, tags), index)
+
+
+    def __str__(self):
+        return 'VerbSemanticsInstance(' + self.sentence + ', ' + str(self.label) + ')'
+
+    def __make_tag_sequence(self, arg1: (int, int), arg2: (int, int)) -> List[str]:
+        """
+        Convert (arg1, arg2) into a tag sequence over the sentence
+        :param arg1: startIndex,endIndex of arg1 in the sentence 
+        :param arg2: startIndex,endIndex of arg2 in the sentence
+        :return: tag sequence
+        E.g. sentence = "Roots absorb water from the soil"
+            arg1 = (4,5)  // the soil
+            arg2 = (0,0)  // roots
+            tag sequence = (B-ARG2, O, O, O, B-ARG1, I-ARG1)
+        """
+        tags = ["O"] * self.sentence.__len__()
+        if arg1[0] != -1:
+            tags[arg1[0]] = "B-" + "ARG1"
+            for index1 in range(arg1[0]+1, arg1[1]):
+                tags[arg1[0]] = "I-" + "ARG1"
+        if arg2[0] != -1:
+            tags[arg2[0]] = "B-" + "ARG2"
+            for index2 in range(arg2[0]+1, arg2[1]):
+                tags[arg2[0]] = "I-" + "ARG2"
+        return tags
+
+    @overrides
+    def words(self) -> Dict[str, List[str]]:
+        """
+        Creates 3 different word dictionaries:
+        1) words in the sentence
+        2) state change types
+        3) tags for arguments
+        """
+        words = self._words_from_text(self.sentence)
+        words['state_changes'] = [self.label[0]]
+        words['tags'] = self.label[1]
+        return words
+
+    @overrides
+    def to_indexed_instance(self, data_indexer: DataIndexer):
+        sentence_indices = self._index_text(self.sentence, data_indexer)
+        verb_indices = sentence_indices[self.verb[0]:self.verb[1]]
```

Comment:
Even in that case, though, you probably don't need to duplicate the word ids - you just pull them out again after they've been embedded, using the binary vector.  If you do it this way, you need to worry about padding the verb and entity indices to their own lengths, and you need to run them through embeddings / encoders separately.